installer.sh

```diff
@@ -0,0 +1,238 @@
+
+## Handle Arguments
+
+if [[ ! -n $1 ]]; then
+  echo "No arguments provided, installing with"
+  echo "default configuration values."
+fi
+
+while [ $# -gt 0 ]; do
+  case "$1" in
+    --travis_worker_version=*)
+      TRAVIS_WORKER_VERSION="${1#*=}"
+      ;;
+    --docker_version=*)
+      DOCKER_VERSION="${1#*=}"
+      ;;
+    --aws=*)
+      AWS="${1#*=}"
+      ;;
+    --travis_enterprise_host=*)
+      TRAVIS_ENTERPRISE_HOST="${1#*=}"
+      ;;
+    --travis_enterprise_security_token=*)
+      TRAVIS_ENTERPRISE_SECURITY_TOKEN="${1#*=}"
+      ;;
+    --travis_enterprise_build_endpoint=*)
+      TRAVIS_ENTERPRISE_BUILD_ENDPOINT="${1#*=}"
+      ;;
+    --travis_queue_name=*)
+      TRAVIS_QUEUE_NAME="${1#*=}"
+      ;;
+    --skip_docker_populate=*)
+      SKIP_DOCKER_POPULATE="${1#*=}"
+      ;;
+    *)
+      printf "*************************************************************\n"
+      printf "* Error: Invalid argument.                                  *\n"
+      printf "* Valid Arguments are:                                      *\n"
+      printf "*  --travis_worker_version=x.x.x                            *\n"
+      printf "*  --docker_version=x.x.x                                   *\n"
+      printf "*  --aws=true                                               *\n"
+      printf "*  --travis_enterprise_host="demo.enterprise.travis-ci.com" *\n"
+      printf "*  --travis_enterprise_security_token="token123"            *\n"
+      printf "*  --travis_enterprise_build_endpoint="build-api"           *\n"
+      printf "*  --travis_queue_name="builds.linux"                       *\n"
+      printf "*  --skip_docker_populate=true                              *\n"
+      printf "*************************************************************\n"
+      exit 1
+  esac
+  shift
+done
+
+if [[ ! -n $DOCKER_VERSION ]]; then
+  export DOCKER_VERSION="17.12.0~ce-0~ubuntu"
+else
+  export DOCKER_VERSION
+fi
+
+if [[ ! -n $TRAVIS_WORKER_VERSION ]]; then
+  export TRAVIS_WORKER_VERSION="v3.5.0"
+else
+  export TRAVIS_WORKER_VERSION
+fi
+
+## We only want to run as root
+root_check() {
+  if [[ $(whoami) != "root" ]]; then
+    echo "This should only be run as root"
+    exit 1
+  fi
+}
+
+root_check
+##
+
+install_packages() {
+  apt-get install -y
+    apt-get install -y apt-transport-https \
+    ca-certificates \
+    curl \
+    software-properties-common \
+    jq
+}
+
+install_packages
+
+## Install and setup Docker
+docker_setup() {
+  : "${DOCKER_CONFIG_FILE:=/etc/default/docker}"
+
+  if [[ ! -f $DOCKER_APT_FILE ]]; then
+    curl -fsSL 'https://download.docker.com/linux/ubuntu/gpg' | apt-key add -
+    add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+  fi
+
+  apt-get update
+
+  if ! docker version &>/dev/null; then
+    apt-get install -y docker-ce=$DOCKER_VERSION
+  fi
+
+  jq -n '{"storage-driver": "overlay2", "icc": false, "log-driver": "journald"}' > /etc/docker/daemon.json
```

Comment:
Are these the defaults, or do we need to set them manually like this still? (can't remember if icc is default). Also what is `jq` doing here?

Reply:
`jq` is ensuring the right format of the json, `icc` was taken over from the old installer as it is, so I'm not sure if we still need that.

Reply:
Thinking back yeah I think `icc` needs to be off for security reasons so jobs can't see other containers. Super cool on `jq`, I mean, it's already installed so may as well use it here.

Reply:
I opened a ticket for `icc`: https://github.com/travis-ci/travis-enterprise-worker-installers/issues/5